### PR TITLE
feat: `_` can be used as separators in environment variable names

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
You can use both as environment variable name:
- `database.default.hostname` (now) - looking for it first
- `database_default_hostname` (new) - if not found, looking for it

See #4026

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPdocs
- [x] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by MGatner** [comment]
The code looks good. Could you share your motivation behind this change? Are there other frameworks that use underscores? Or perhaps deployment solutions? Or do you have an underscore fetish? 😉

**Comment by samsonasik** [comment]
never use `_` usage, what if user has existing `_` config as a part of value? 

```php
SimpleConfig.onedeep_value=baz
```

as `onedeep_value` as the key itself. is that previously working or after this change, that will still works?

**Comment by kenjis** [comment]
@MGatner Some environments can't use environment variables with dot.
See #4026

And I saw this tweet:
https://twitter.com/eclairtom/status/1443603184597962752
English Translation:
> Versioned up to #Docker desctop 4.1.0. docker-compose is no longer usable.
The .env used by CodeIgniter4 is now affected. Because it uses constants that contain dots. (4.0 was ignored.)
I have to change the folder structure for docker at worst.

**Comment by kenjis** [comment]
@samsonasik Good question!
I think property name like `onedeep_value` is not used by CI4,
but users might use it.

I added a test case, and it passed.

File: tests/system/Config/fixtures/.env [hunk 1]
```diff
@@ -6,15 +6,20 @@ NULL=
 
 SimpleConfig.onedeep=baz
 SimpleConfig.default.name=ci4
-SimpleConfig.simple.name=complex
+# Use underscore as separator
+SimpleConfig_simple_name=complex
+# Properity name with undersocre
+SimpleConfig.onedeep_value=bar
```
Comment (samsonasik) [inline tests/system/Config/fixtures/.env:12]: I think another case is needed, eg

```
SimpleConfig.one_deep.under_deep
```

for user use "." but use underscored keys

Comment (kenjis) [reply tests/system/Config/fixtures/.env]: I added a test.

**Review by michalsn — APPROVED** [review]
I think this is probably the right direction to solve the problem with dot variables for some environments.

The only thing missing is documentation. We don't want this to become a secret feature, so it needs to be documented in the user guide.

**Comment by kenjis** [comment]
@michalsn Thanks! I added it to the documentation.

**Review by MGatner — APPROVED** [review]
Nice teamwork! This looks ready to me.

**Pull request merged**